flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/AbstractFileSource.java

```diff
@@ -177,6 +180,24 @@ public TypeInformation<T> getProducedType() {
         return readerFormat.getProducedType();
     }
 
+    @Override
+    public int inferParallelism(Context dynamicParallelismContext) {
```

Comment:
`AbstractFileSource` is a base for both `FileSource` and `HiveSource`. If we only target `FileSource` in the PR, why not directly modify `FileSource`?